chainerx_cc/chainerx/routines/creation.cc

```diff
@@ -337,4 +338,66 @@ Array Linspace(
     return out;
 }
 
+Array Tri(int64_t n, absl::optional<int64_t> m, absl::optional<int64_t> k, absl::optional<Dtype> dtype, Device& device) {
+    if (!m.has_value()) {
+        m = n;
+    }
+    if (!k.has_value()) {
+        k = 0;
+    }
+    if (!dtype.has_value()) {
+        dtype = Dtype::kFloat64;
+    }
+    if (n < 0 || m < 0) {
+        throw DimensionError{"Negative dimensions are not allowed"};
+    }
```

Comment:
Can you align this behavior to NumPy instead of throwing errors? You might have to be careful with setting the proper shapes for 0-sized arrays. If it takes too much time, it's okay to leave it as a TODO, but in that case, please write a TODO comment.

Reply:
I think NumPy's behavior is a bug in the implementation. For example `numpy.eye` and other array creation routines raise an error for negative dimensions.
CuPy also raises an error for negative dimensions for `cupy.tri`

Reply:
Hm, you are right. Thanks for the insight. I think the error in CuPy is a coincidence because of the usage of `cp.empty(<negative dims to yield error>)` but it might be fine to leave it as is. We can leave this behavior undocumented but maybe add a comment in the code that this is intentional.

Reply:
I've added a comment on that in the code.